src/elements/Button/Button.js

```diff
@@ -1,33 +1,136 @@
+    useKeyOnly(toggle, 'toggle'),
+    'button',
+    className
+  )
+
+  const rest = getUnhandledProps(Button, props)
+
+  return (
+    <button type={type} className={classes} {...rest}>
+      {iconPropRenderer(icon)}
+      {children}
+    </button>
+  )
+}
+
+Button.Group = ButtonGroup
+
+Button._meta = {
+  library: META.library.semanticUI,
+  name: 'Button',
+  type: META.type.element,
+  props: {
+    attached: ['left', 'right', 'top', 'bottom'],
+    color: sui.colors,
+    floated: sui.floats,
+    labeled: ['left icon', 'right icon'],
+    size: sui.sizes,
+  },
+}
+
+Button.propTypes = {
+  /** A button can show it is currently the active user selection */
+  active: PropTypes.bool,
+
+  /** A button can animate to show hidden content */
+  animated: PropTypes.bool,
+
+  /** A button can be attached to the top or bottom of other content */
+  attached: PropTypes.oneOf(Button._meta.props.attached),
+
+  /** A basic button is less pronounced */
+  basic: PropTypes.bool,
+
+  /** Primary content of the button */
+  children: PropTypes.node,
+
+  /** A button can be circular */
+  circular: PropTypes.bool,
 
-export default class Button extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-  }
-
-  static defaultProps = {
-    type: 'button',
-  }
-
-  static _meta = {
-    library: META.library.semanticUI,
-    name: 'Button',
-    type: META.type.element,
-  }
-
-  render() {
-    const classes = classNames(
-      'ui',
-      this.props.className,
-      'button'
-    )
-    return (
-      <button {...this.props} className={classes}>
-        {this.props.children}
-      </button>
-    )
-  }
+  /** Classes to add to the button className. */
+  className: PropTypes.string,
+
+  /** A button can have different colors */
+  color: PropTypes.oneOf(Button._meta.props.color),
+
+  /** A button can reduce its padding to fit into tighter spaces */
+  compact: PropTypes.bool,
+
+  /** A button can show it is currently unable to be interacted with */
+  disabled: PropTypes.bool,
+
+  /** A button can be aligned to the left or right of its container */
+  floated: PropTypes.oneOf(Button._meta.props.floated),
+
+  /** A button can take the width of its container */
+  fluid: PropTypes.bool,
+
+  /** Add an icon by icon className or pass an <Icon /.>*/
+  icon: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.element,
+  ]),
+
+  /** A button can be formatted to appear on dark backgrounds */
+  inverted: PropTypes.bool,
+
+  /** A button can appear alongside a label */
+  labeled: PropTypes.oneOf(Button._meta.props.labeled),
+
+  /** A button can show a loading indicator */
+  loading: PropTypes.bool,
+
+  /** A button can be formatted to toggle on and off */
+  toggle: PropTypes.bool,
+
+  /**  */
+  type: PropTypes.string,
+
+  /** A button can have different sizes */
+  size: PropTypes.oneOf(Button._meta.props.size),
 }
+
+Button.defaultProps = {
+  type: 'button',
+}
```

Comment:
Let's pull the type prop and it's default value.  The user can set this if they wish.